### PR TITLE
Nsdl interface list accesses

### PR DESCRIPTION
**Pull request by kuggenhoffen**
Fix issue where it is possible to modify __nsdl_interface_list while someone else is accessing it.

### CODE REVIEW

**Comment by kuggenhoffen** [comment]
@anttiylitokola @TeroJaasko

**Comment by kuggenhoffen** [comment]
This PR depends on https://github.com/ARMmbed/mbed-client-c/pull/96

File: source/nsdlaccesshelper.cpp [hunk 1]
```diff
@@ -31,7 +28,7 @@ uint8_t __nsdl_c_callback(struct nsdl_s *nsdl_handle,
                           sn_nsdl_capab_e nsdl_capab)
 {
     uint8_t status = 0;
-    M2MNsdlInterface  *interface = get_interface(nsdl_handle);
+    M2MNsdlInterface *interface = (M2MNsdlInterface*)sn_nsdl_get_context(nsdl_handle);
```
Comment (TeroJaasko) [inline source/nsdlaccesshelper.cpp]: I suppose a "assert(interface)" would make it sure the context is set properly on all cases.

**Comment by TeroJaasko** [comment]
+1

**Comment by jankii01** [comment]
Dependency on mbed-client-c

**Comment by kuggenhoffen** [comment]
Need to wait next release where we can use new non-backward compatible mbed-client-c

**Comment by kuggenhoffen** [comment]
Combined with #379 into #399

**Pull request closed without merging**